tests/integration/sync_test.go

```diff
@@ -5183,3 +5185,124 @@ func Test_Sync_ConsumerGroupConsumerFromUpstream(t *testing.T) {
+		t.Skip("Skipping because environment variable KONG_LICENSE_DATA not found")
+	}
+	setup(t)
+
+	buf, err := os.ReadFile("testdata/sync/032-licenses/config-with-license.yaml")
+	require.NoError(t, err)
+	fileContent := strings.ReplaceAll(string(buf), "__KONG_LICENSE_DATA__", fmt.Sprintf("'%s'", kongLicensePayload))
+	configFile, err := os.CreateTemp("/tmp", "kong-license-test")
+	require.NoError(t, err)
+	defer os.Remove(configFile.Name())
+
+	os.WriteFile(configFile.Name(), []byte(fileContent), os.ModeTemporary)
+	client, err := getTestClient()
+	ctx := context.Background()
+
+	t.Run("create_license_and_dump_results", func(t *testing.T) {
+		currentState, err := fetchCurrentState(ctx, client, deckDump.Config{IncludeLicenses: true})
+		require.NoError(t, err)
+
+		targetState := stateFromFile(ctx, t, configFile.Name(), client, deckDump.Config{
+			IncludeLicenses: true,
+		})
+		syncer, err := deckDiff.NewSyncer(deckDiff.SyncerOpts{
+			CurrentState: currentState,
+			TargetState:  targetState,
+
+			KongClient:      client,
+			IncludeLicenses: true,
+		})
+		require.NoError(t, err)
+
+		require.NoError(t, err, "Should get test client")
+		stats, errs, changes := syncer.Solve(ctx, 1, false, true)
+		require.Len(t, errs, 0, "Should have no errors in syncing")
+		logEntityChanges(t, stats, changes)
+
+		newState, err := fetchCurrentState(ctx, client, deckDump.Config{IncludeLicenses: true})
+		require.NoError(t, err)
+
+		licenses, err := newState.Licenses.GetAll()
+		require.NoError(t, err)
+		// Avoid dumping of `licenses` to leak sensitive content.
+		require.Equal(t, 1, len(licenses))
+		// Compare hashes to avoid content of licenses to be leaked.
+		expectedLicenseHash := sha1.Sum([]byte(kongLicensePayload))
+		actualLicenseHash := sha1.Sum([]byte(*licenses[0].Payload))
+		require.Equal(t, expectedLicenseHash, actualLicenseHash, "Hash of license payload should be the same as env KONG_LICENSE_DATA")
+	})
+
+	t.Run("dump_with_includeLicense_disabled", func(t *testing.T) {
+		stateWithoutLicenses, err := fetchCurrentState(ctx, client, deckDump.Config{IncludeLicenses: false})
+		require.NoError(t, err)
+		licenses, err := stateWithoutLicenses.Licenses.GetAll()
+		require.NoError(t, err)
+		require.Equal(t, 0, len(licenses))
+	})
+
+	t.Run("sync_with_includeLicenses_false", func(t *testing.T) {
+		currentState, err := fetchCurrentState(ctx, client, deckDump.Config{IncludeLicenses: true})
+		require.NoError(t, err)
+		stateWithoutLicense := stateFromFile(ctx, t,
+			"testdata/sync/032-licenses/config-without-license.yaml",
+			client,
+			deckDump.Config{IncludeLicenses: true},
+		)
+		syncer, err := deckDiff.NewSyncer(deckDiff.SyncerOpts{
+			CurrentState: currentState,
+			TargetState:  stateWithoutLicense,
+
+			KongClient:      client,
+			IncludeLicenses: false,
+		})
+		require.NoError(t, err)
+
+		stats, errs, changes := syncer.Solve(ctx, 1, false, true)
+		require.Len(t, errs, 0, "Should have no errors in syncing")
```

Comment:
FYI there's also `Empty()` if you'd like to use it: https://pkg.go.dev/github.com/stretchr/testify/require#Empty

Reply:
Done